lib/smb2/file.rb

```diff
@@ -58,23 +79,23 @@ def inspect
   # @param length [Fixnum,nil] number of bytes to read. If this is nil, read
   #   the whole file, i.e. {#size} bytes.
   # @return [String] full contents of the remote file
-  def read(offset: 0, length: nil)
+  def read(length = nil, offset: self.pos)
```

Comment:
Yeah, that seems pretty clunky, and would definitely make the method overly complex. What about abandoning the offset and only allowing `read()` (to get everything from current `pos` to end), and `read(length)` to get a certain number of bytes starting from `pos`?